### PR TITLE
Stream message dispatcher for invoking requests to annotated methods.

### PR DESCRIPTION
**Pull request by johnou**

### CODE REVIEW

**Comment by JoeHegarty** [comment]
Seems reasonable to me, @DanielSperry ?

File: actors/test/actor-tests/src/test/java/com/ea/orbit/actors/test/streams/StreamMessageDispatcherTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    @SuppressWarnings("Duplicates")
+    @Test(timeout = 30_000L)
+    public void testStreamMessageDispatcher()
+    {
+        createStage();
+        StreamMessageDispatcher messageDispatcher = new StreamMessageDispatcher(Collections.singleton(this));
+        try
+        {
+            AsyncStream<Object> test = AsyncStream.getStream(Object.class, "test");
+            test.subscribe((d, t) -> {
```
Comment (johnou) [inline actors/test/actor-tests/src/test/java/com/ea/orbit/actors/test/streams/StreamMessageDispatcherTest.java]: We could provide some wrapper for dealing with subscription in the constructor or some builder / factory?

**Comment by DanielSperry** [comment]
It's an util class, right?
looks good.

Thanks, @johnou

**Comment by johnou** [comment]
Yep, use case is that I have an actor which will receive several different message types over a stream and having a bunch of instance of checks didn't seem desirable.

**Pull request merged**